lib/fetch.js

```diff
@@ -138,6 +147,40 @@ async function stream(method, url, options) {
     }
 }
 
+/**
+ * Uses this file's "stream" method to submit an HTTP request. If the response to
+ * the request has a body that is a stream, this method will use the stream's
+ * events to provide the response to the request.
+ *
+ * If the response _doesn't_ provide a streamed body, the method will return
+ * the response as-is.
+ *
+ * The purpose of this method is primarily to process the response differently,
+ * depending on whether it was initiated from Node.JS or from a browser.
+ *
+ * @param {String} method HTTP method of the request to submit.
+ * @param {String} url URL of the request to submit.
+ * @param {Object} options Additional options to provide to fetch when
+ *  submitting the request.
+ * @returns {*} HTTP response of the request.
+ */
+async function handleStream(method, url, options) {
+    const response = await stream(method, url, options);
+    if (response.body && response.body.on) {
```

Comment:
Can response be undefined/null?
```suggestion
    if (response && response.body && response.body.on) {
```

Reply:
I think we'd have bigger problems downstream if it were, but I'll add the check anyway to be safe.